src/librustc_save_analysis/lib.rs

```diff
@@ -144,7 +144,23 @@ impl<'l, 'tcx> SaveContext<'l, 'tcx> {
                     span: self.span_from_span(item.ident.span),
                     name: item.ident.to_string(),
                     qualname,
-                    value: make_signature(decl, generics),
+                    value: fn_to_string(
+                        decl,
+                        hir::FnHeader {
+                            // functions in extern block are implicitly unsafe
+                            unsafety: hir::Unsafety::Unsafe,
+                            // functions in extern block cannot be const
+                            constness: hir::Constness::NotConst,
+                            abi: self.tcx.hir().get_foreign_abi(item.hir_id),
```

Comment:
I managed to get the real abi of the extern block using `get_foreign_abi`. I don't think we should change the hir to store a `FnHeader` in `ForeignItem::Fn`. I tried it locally, it works, but its really invasive and somehow orthogonal to this change.

Reply:
I think you're right, let's stick with what we have. Thanks!